src/components/map/map.e2e.ts

```diff
@@ -19,22 +20,78 @@ describe("gx-map", () => {
   });
 
   it("should set a given center", async () => {
-    element.setProperty("center", "38.87097161910191, -77.0559650659561");
+    await page.setContent(
+      "<gx-map center='38.87097161910191, -77.0559650659561'></gx-map>"
+    );
+    await page.waitForChanges();
+    element = await page.find("gx-map");
     await page.waitForChanges();
     expect(await element.getProperty("center")).toEqual(
       "38.87097161910191, -77.0559650659561"
     );
   });
 
-  it("should set a given maxZoom", async () => {
-    element.setProperty("maxZoom", 21);
+  it("should set a default maxZoom if given maxZoom exceed optimal range", async () => {
+    await page.setContent("<gx-map max-zoom='24'></gx-map>");
+    await page.waitForChanges();
+    element = await page.find("gx-map");
     await page.waitForChanges();
-    expect(await element.getProperty("maxZoom")).toEqual(21);
+    expect(await element.getProperty("maxZoom")).toEqual(20);
   });
 
-  it("should set a given zoom", async () => {
-    element.setProperty("zoom", 16);
+  it("should set a default zoom if given zoom exceed optimal range", async () => {
+    await page.setContent("<gx-map zoom='35'></gx-map>");
+    await page.waitForChanges();
+    element = await page.find("gx-map");
+    await page.waitForChanges();
+    expect(await element.getProperty("zoom")).toEqual(19);
+  });
+
+  it("should set the default mapType if mapType is no defined by user", async () => {
+    await page.setContent("<gx-map></gx-map>");
+    await page.waitForChanges();
+    element = await page.find("gx-map");
+    expect(await element.getProperty("mapType")).toEqual("standard");
+  });
+
+  it("should set the mapType defined by user", async () => {
+    await page.setContent("<gx-map map-type='satellite'></gx-map>");
+    await page.waitForChanges();
+    element = await page.find("gx-map");
+    expect(await element.getProperty("mapType")).toEqual("satellite");
+  });
```

Comment:
You did a very nice job with the tests. However, I think this one test isn't really testing your programming but Stencil's ability to map the component's attributes to properties. I think we could leave this one out.

Reply:
As u wish. Done in 6eb93ae